### PR TITLE
wiremock header validation

### PR DESCRIPTION
**Pull request by spetrychenko**

### CODE REVIEW

File: xs2a-adapter-service-impl/src/test/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptorTest.java [hunk 1]
```diff
@@ -58,6 +58,24 @@ void postHandle_allMatch() throws JsonProcessingException {
             .doesNotContainKey(ResponseHeaders.X_GTW_ASPSP_CHANGES_DETECTED);
     }
 
+    @Test
+    void postHandle_ContentTypeNotMatching() throws JsonProcessingException {
+        RequestBuilderImpl request = new RequestBuilderImpl(httpClient, "POST", "https://bank.com/v1/consents");
```
Comment (berezkin88) [inline xs2a-adapter-service-impl/src/test/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptorTest.java:63]: There is a constant for POST consent url - CONSENTS_URL

File: xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java [hunk 2]
```diff
@@ -183,12 +184,43 @@ public WiremockStubDifferenceDetectingInterceptor(Aspsp aspsp) {
...
+    private Optional<String> analyzeRequestHeaders(Request.Builder builder, Map<String, Object> stubHeaders) {
+        Map<String, String> currentHeaders = builder.headers();
+        Sets.SetView<String> headers = Sets.intersection(WiremockSupportedHeader.set(), currentHeaders.keySet());
+        Sets.SetView<String> presentHeaders = Sets.intersection(headers, stubHeaders.keySet());
+        Sets.SetView<String> requestDiff = Sets.difference(headers, stubHeaders.keySet());
+        Sets.SetView<String> stubDiff = Sets.difference(stubHeaders.keySet(), headers);
+        StringBuilder differences = new StringBuilder();
+        presentHeaders.forEach(key -> {
+            WiremockSupportedHeader supportedHeader = WiremockSupportedHeader.resolve(key).get();
+            if (!supportedHeader.isEqual(getStubRequestHeader(stubHeaders, key), currentHeaders.get(key))) {
```
Comment (berezkin88) [inline xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:196]: `isEqual` always returns _true_, except for Content-Type header

Comment (berezkin88) [reply xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:196]: Is it an expected behavior for `isEqual` always to return _true_, except for cases where the value must be verified?

Comment (spetrychenko) [reply xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java:196]: yes. for now, we compare the value of the header only for Content-Type header. For other headers, the comparison is ignored.

File: xs2a-adapter-service-impl/src/test/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptorTest.java (hunk 1 above)
Comment (spetrychenko) [reply xs2a-adapter-service-impl/src/test/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptorTest.java]: fixed, thanks

**Review by berezkin88 — APPROVED** [review]

**Pull request merged**